src/RefractAPI.cc

```diff
@@ -399,23 +461,26 @@ namespace drafter {
         return element;
     }
 
-    refract::IElement* ResourceToRefract(const snowcrash::Resource& resource)
+    refract::IElement* ResourceToRefract(const SectionInfo<snowcrash::Resource>& resource)
     {
         refract::ArrayElement* element = new refract::ArrayElement;
         RefractElements content;
 
         element->element(SerializeKey::Resource);
-        element->meta[SerializeKey::Title] = refract::IElement::Create(resource.name);
-        element->attributes[SerializeKey::Href] = refract::IElement::Create(resource.uriTemplate);
 
-        if (!resource.parameters.empty()) {
-            element->attributes[SerializeKey::HrefVariables] = ParametersToRefract(resource.parameters);
+        element->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_SECTION_INFO(resource, name));
+
+        element->attributes[SerializeKey::Href] = PrimitiveToRefract(MAKE_SECTION_INFO(resource, uriTemplate));
+
+        if (!resource.section.parameters.empty()) {
+            element->attributes[SerializeKey::HrefVariables] = ParametersToRefract(MAKE_SECTION_INFO(resource, parameters));
         }
 
-        content.push_back(CopyToRefract(resource.description));
-        content.push_back(DataStructureToRefract(resource.attributes));
+        content.push_back(CopyToRefract(MAKE_SECTION_INFO(resource, description)));
+        content.push_back(DataStructureToRefract(MAKE_SECTION_INFO(resource, attributes)));
 
-        std::transform(resource.actions.begin(), resource.actions.end(), std::back_inserter(content), ActionToRefract);
+        SectionInfoCollection<snowcrash::Actions> actions(resource.section.actions, resource.sourceMap.actions);
+        std::transform(actions.begin(), actions.end(), std::back_inserter(content), ActionToRefract);
```

Comment:
I mean it is not good idea

Take a look at original code:

``` 
SectionInfoCollection<snowcrash::Actions> actions(resource.section.actions, resource.sourceMap.actions);
std::transform(actions.begin(), actions.end(), std::back_inserter(content), ActionToRefract);
```
As input param to macro you need:
- type of collection `snowcrash::Actions`
- input collection `resource.section.actions`
- input sourcemap `resource.sourceMap.actions`
- output colection `content`
- transformation function `ActionToRefract`

So macro will looks like:
```
TRANSFORM_MACRO(snowcrash::Actions, resource.section.actions, resource.sourceMap.actions, content, ActionToRefract);
```

We can simplify duplicity fetching parts `section` and `sourceMap` like in `MAKE_SECTION_INFO()`

But it will convert macro into form:
```
TRANSFORM_MACRO(snowcrash::Actions, resource, actions, content, ActionToRefract);
``` 

But it absolutely does not explain what happens there. Two lines of original code are better readable than this macro.

Possible solutions I see:
- add macro like `MAKE_SECTION_INFO()` for collections, or
- replace macro `MAKE_SECTION_INFO()` by macro usable for both `SectionInfo` and `SectionInfoCollection`

```
#define SECTION_INFO(from, part) from.section.part, from.sourceMap.part
```